main/modes/mfpaint/mode_paint.c

```diff
@@ -205,6 +204,12 @@ void paintMenuInitialize(void)
 
 void paintSetupMainMenu(void)
 {
+    if (paintMenu->menu != NULL)
+    {
+        deinitMenu(paintMenu->menu);
+    }
```

Comment:
`deinitMenu()` doesn't set `paint->menu` to `NULL`, so maybe it's possible to double-free here? I didn't check the rest of the code paths.

Reply:
I think this should be covered by the other paths, but no reason not to add it anyway.